core/vm/evm.go

```diff
@@ -65,6 +66,10 @@ func (evm *EVM) precompile(addr common.Address) (contract.StatefulPrecompiledCon
 	if !ok {
 		return nil, false
 	}
+	// Precompile is registered, check if it's enabled at this block height
+	if !slices.Contains(evm.enabledPrecompiles, addr) {
+		return nil, false
+	}
```

Comment:
Could be moved above the `modules.GetPrecompileModuleByAddress(addr)`, as it only need to be fetched if it's enabled

Reply:
I like the encapsulation, but I'm actually leaning towards keeping it

Reply:
My reasoning being is that enabled precompiles are a concept only ethermint and the core vm.  The precompile registry has no concept of enabled and is only used to map addresses to precompile code.

Extending the registry to support that feature feels like core/evm is sharing it's responsiblity and creating a subtle coupling.

Reply:
Thoughts @drklee3 @evgeniy-scherbina ?

Maybe an alternative could be a private function on the evm type? `evm.IsStatefulPrecompileEnabled(addr)`?

Reply:
@nddeluca you mean you're thinking about encapsulating `enabledPrecompiles` feature inside `registry` package?

I just don't know how to do it in subtle way, registry is global and doesn't have any context, but `enabledPrecompiles` changes with each new tx/query.

Reply:
@nddeluca you mean using `evm.IsStatefulPrecompileEnabled(addr)` instead of `!slices.Contains(evm.enabledPrecompiles, addr)`?

Reply:
Ya, though I mistyped, probably want lower case `evm.isStatefulPrecompileEnabled` -- I don't think it needs to be on the public interface.

Also, though I think Derrick's point still stands.  We could check the enabled list first, however I'm unsure which results in better performance.  My guess is the O(1) map lookup is first is the best approach, the search is O(N) (tho the slice is sorted, so could be O(logn)) and in for the happy path this works well since most transactions will be for enabled contracts, and unenabled contracts don't result in a N or LogN search.

If checking enabledPrecompiles was O(1), then I'd see a strong case for checking it first.

Reply:
as of now I think it doesn't affect performance, both checks are O(N)

`GetPrecompileModuleByAddress` - can be easily optimized to O(logN) with binary search

regarding `enabledPrecompiles` I can replace it with hash-map, but I think performance impact will be negligible? Because we recreate EVM on every transaction, and I think it's much heavier operation.

Reply:
Fixed according to discussion

Reply:
Ahh, right @evgeniy-scherbina you are correct. I don't know why I thought it was O(1), I think I got confused by the core precompiles which are looked up by map

Reply:
Created #29 to track the binary search improvement

Reply:
I agree, it's likely negligible, so not worth pursuing unless it becomes a bottleneck